test/stdlib/POSIX.swift

```diff
@@ -17,21 +13,59 @@ chdir(CommandLine.arguments[1])
 var POSIXTests = TestSuite("POSIXTests")
 
 let semaphoreName = "TestSem"
+#if os(Android)
+// In Android, the cwd is the root directory, which is not writable.
+let fn: String = {
+  let capacity = Int(PATH_MAX)
+  let resolvedPath = UnsafeMutablePointer<Int8>.allocate(capacity: capacity)
+  resolvedPath.initialize(repeating: 0, count: capacity)
+  defer {
+    resolvedPath.deinitialize(count: capacity)
+    resolvedPath.deallocate()
+  }
+  guard let _ = realpath("/proc/self/exe", resolvedPath) else {
+    fatalError("Couldn't obtain executable path")
+  }
+
+  let length = strlen(resolvedPath)
+  guard length != 0 else {
+    fatalError("Couldn't obtain valid executable path")
+  }
+
+  // Search backwards for the last /, and turn it into a null byte.
+  for idx in (0..<length).reversed() {
```

Comment:
Honestly I can't tell which version simpler to read, but reversing a range looks a bit odd. I'll just mention it here, feel free to ignore :-)

```suggestion
  for idx in stride(from: length-1, through: 0, by: -1) {
```